example/linux/gw-example/main.c

```diff
@@ -101,11 +101,11 @@ static bool MQTT_publish(char * topic, uint8_t * payload, size_t payload_size, b
 static void * publish_thread(void * unused)
 {
     message_to_publish_t * message;
-    pthread_mutex_lock(&m_queue_mutex);
```

Comment:
It looks similar as it's done in dispatch_indication : don't we have also a problem there ? In the case we go to continue, the mutex is unlock when pthread_cond_wait. A possible correction could be to remove pthread_mutex_unlock ligne 100 in file plateform.c, or we can do the same correction as in publish_thread